pkg/proxy/handler.go

```diff
@@ -158,5 +131,37 @@ func newReverseProxy(host string) *httputil.ReverseProxy {
 			AdditionalProxyRewrite(r)
 		},
 	}
-	return proxy
+
+	proxy.ModifyResponse = func(r *http.Response) error {
+		if h.isRetryCode(r.StatusCode) {
+			// Returning an error will trigger the ErrorHandler.
+			return errors.New("retry")
+		}
+		return nil
+	}
+
+	proxy.ErrorHandler = func(w http.ResponseWriter, r *http.Request, err error) {
+		if err != nil && pr.attempt < h.MaxRetries {
```

Comment:
This retries on any error. Is this correct?

Reply:
My understanding is this will retry on any error related to reaching / connecting to the backend by default (for example errors returned by `http.Get()`). For status code errors it will only retry on those reported in `.ModifyResponse`. I think this is the desired behavior. Thoughts?

Reply:
the doc specifies it `that handles errors reaching the backend or errors from ModifyResponse.` Looks good